### ISSUE TITLE
Make it possible to change the theme variant on the fly

### ISSUE THREAD

**Issue by Legioth** [vaadin/flow#4575]
In practice, this would mean that there should be an easy way of triggering `AbstractTheme.getBodyAttributes(String)` again and automatically applying the returned attributes to the `<body>`, while maybe also removing any attributes that have previously been set by the same mechanism.

**Comment by Legioth** [comment]
As a workaround, you can use `executeJavaScript` to manually change attributes on `<body>`, e.g. something like `UI.getCurrent().getPage().executeJavaScript("document.body.setAttribute('theme', 'dark')");`.

**Comment by roomtek** [comment]
you can theme on the fly by getting the element of the currentUI and setting the attribute

`UI.getCurrent().getElement().setAttribute("theme", "dark");`

**Comment by muksihs** [comment]
@jouni I have to add "style='height: 100%;'" to the HTML tag to get the viewport to fill the display area.

**Comment by jouni** [comment]
@muksihs, yes, that’s the downside of using `theme="dark"` on the `<body>` element. You can add it to the `<html>` element to avoid setting any heights.

Another solution is to use `min-height: 100vh;` for the body.

**Comment by Legioth** [comment]
I think this was indirectly implemented recently. Open a new issue if something is still missing.
